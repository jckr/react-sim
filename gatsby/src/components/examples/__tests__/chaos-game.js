import React from 'react';
import renderer from 'react-test-renderer';
import {
  Frame,
  CustomControls,
  init,
  updateData,
  updateAttractors,
} from '../chaos-game';
import Random from '../../helpers/random';

describe('chaos-game', () => {
  const params = {
    height: 500,
    width: 500,
    nbAttractors: 7,
    angle: 0,
    r: 1,
    rules: '1001100',
  };
  let data;
  it('data function works', () => {
    const rng = new Random();
    data = init(params, rng.get);
    expect(data).toMatchSnapshot();
    for (let tick = 1; tick < 5; tick++) {
      data = updateData({ data, tick, params }, rng.get);
      expect(data).toMatchSnapshot();
    }
  });
  it('model renders properly', () => {
    const controls = renderer
      .create(<CustomControls params={params} setParams={jest.fn()} />)
      .toJSON();
    expect(controls).toMatchSnapshot();
    const tree = renderer
      .create(<Frame data={data} params={params} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
