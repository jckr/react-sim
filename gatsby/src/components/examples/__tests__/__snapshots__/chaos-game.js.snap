// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chaos-game data function works 1`] = `
Object {
  "attractors": Array [
    Object {
      "x": 250.00000000000003,
      "y": 12.5,
    },
    Object {
      "x": 64.31502291384294,
      "y": 101.92117205855078,
    },
    Object {
      "x": 18.454620856816888,
      "y": 302.84872181462464,
    },
    Object {
      "x": 146.9526119595799,
      "y": 463.9801061268245,
    },
    Object {
      "x": 353.04738804042,
      "y": 463.9801061268246,
    },
    Object {
      "x": 481.5453791431831,
      "y": 302.8487218146247,
    },
    Object {
      "x": 435.6849770861571,
      "y": 101.92117205855084,
    },
  ],
  "background": "hsla(351, 30%, 7%, 1)",
  "color": "hsla(272, 77%, 45%, 1)",
  "points": Array [
    Object {
      "x": 487.13412840879755,
      "y": 100.04029933611491,
    },
  ],
  "prevDirection": 0,
}
`;

exports[`chaos-game data function works 2`] = `
Object {
  "attractors": Array [
    Object {
      "x": 250.00000000000003,
      "y": 12.5,
    },
    Object {
      "x": 64.31502291384294,
      "y": 101.92117205855078,
    },
    Object {
      "x": 18.454620856816888,
      "y": 302.84872181462464,
    },
    Object {
      "x": 146.9526119595799,
      "y": 463.9801061268245,
    },
    Object {
      "x": 353.04738804042,
      "y": 463.9801061268246,
    },
    Object {
      "x": 481.5453791431831,
      "y": 302.8487218146247,
    },
    Object {
      "x": 435.6849770861571,
      "y": 101.92117205855084,
    },
  ],
  "background": "hsla(351, 30%, 7%, 1)",
  "color": "hsla(272, 77%, 45%, 1)",
  "points": Array [
    Object {
      "x": 487.13412840879755,
      "y": 100.04029933611491,
    },
    Object {
      "x": 420.09075822460875,
      "y": 282.01020273146975,
    },
  ],
  "prevDirection": 4,
}
`;

exports[`chaos-game data function works 3`] = `
Object {
  "attractors": Array [
    Object {
      "x": 250.00000000000003,
      "y": 12.5,
    },
    Object {
      "x": 64.31502291384294,
      "y": 101.92117205855078,
    },
    Object {
      "x": 18.454620856816888,
      "y": 302.84872181462464,
    },
    Object {
      "x": 146.9526119595799,
      "y": 463.9801061268245,
    },
    Object {
      "x": 353.04738804042,
      "y": 463.9801061268246,
    },
    Object {
      "x": 481.5453791431831,
      "y": 302.8487218146247,
    },
    Object {
      "x": 435.6849770861571,
      "y": 101.92117205855084,
    },
  ],
  "background": "hsla(351, 30%, 7%, 1)",
  "color": "hsla(272, 77%, 45%, 1)",
  "points": Array [
    Object {
      "x": 487.13412840879755,
      "y": 100.04029933611491,
    },
    Object {
      "x": 420.09075822460875,
      "y": 282.01020273146975,
    },
    Object {
      "x": 242.20289056922584,
      "y": 191.96568739501026,
    },
  ],
  "prevDirection": 1,
}
`;

exports[`chaos-game data function works 4`] = `
Object {
  "attractors": Array [
    Object {
      "x": 250.00000000000003,
      "y": 12.5,
    },
    Object {
      "x": 64.31502291384294,
      "y": 101.92117205855078,
    },
    Object {
      "x": 18.454620856816888,
      "y": 302.84872181462464,
    },
    Object {
      "x": 146.9526119595799,
      "y": 463.9801061268245,
    },
    Object {
      "x": 353.04738804042,
      "y": 463.9801061268246,
    },
    Object {
      "x": 481.5453791431831,
      "y": 302.8487218146247,
    },
    Object {
      "x": 435.6849770861571,
      "y": 101.92117205855084,
    },
  ],
  "background": "hsla(351, 30%, 7%, 1)",
  "color": "hsla(272, 77%, 45%, 1)",
  "points": Array [
    Object {
      "x": 487.13412840879755,
      "y": 100.04029933611491,
    },
    Object {
      "x": 420.09075822460875,
      "y": 282.01020273146975,
    },
    Object {
      "x": 242.20289056922584,
      "y": 191.96568739501026,
    },
    Object {
      "x": 297.62513930482294,
      "y": 327.9728967609174,
    },
  ],
  "prevDirection": 4,
}
`;

exports[`chaos-game data function works 5`] = `
Object {
  "attractors": Array [
    Object {
      "x": 250.00000000000003,
      "y": 12.5,
    },
    Object {
      "x": 64.31502291384294,
      "y": 101.92117205855078,
    },
    Object {
      "x": 18.454620856816888,
      "y": 302.84872181462464,
    },
    Object {
      "x": 146.9526119595799,
      "y": 463.9801061268245,
    },
    Object {
      "x": 353.04738804042,
      "y": 463.9801061268246,
    },
    Object {
      "x": 481.5453791431831,
      "y": 302.8487218146247,
    },
    Object {
      "x": 435.6849770861571,
      "y": 101.92117205855084,
    },
  ],
  "background": "hsla(351, 30%, 7%, 1)",
  "color": "hsla(272, 77%, 45%, 1)",
  "points": Array [
    Object {
      "x": 487.13412840879755,
      "y": 100.04029933611491,
    },
    Object {
      "x": 420.09075822460875,
      "y": 282.01020273146975,
    },
    Object {
      "x": 242.20289056922584,
      "y": 191.96568739501026,
    },
    Object {
      "x": 297.62513930482294,
      "y": 327.9728967609174,
    },
    Object {
      "x": 325.3362636726215,
      "y": 395.97650144387103,
    },
  ],
  "prevDirection": 4,
}
`;

exports[`chaos-game model renders properly 1`] = `
<div
  className="css-l3rx45"
>
  <div
    className="css-l3rx45"
  >
    <label
      className="css-knfhfe"
      htmlFor="Vertices"
    >
      Vertices: 
      7
    </label>
    <div
      className="css-190meoe"
    >
      <div
        className="css-1kfmgte"
        onClick={[Function]}
      >
        3
      </div>
      <input
        className="css-1u08zt6"
        id="Vertices"
        max={20}
        min={3}
        name="Vertices"
        onChange={[Function]}
        step={1}
        type="range"
        value={7}
      />
      <div
        className="css-89g11l"
        onClick={[Function]}
      >
        20
      </div>
    </div>
  </div>
  <div
    className="css-l3rx45"
  >
    <div
      className="css-1jq4bbb"
    >
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="1st vertex"
        >
          1st vertex
        </label>
        <button
          aria-checked={true}
          className="css-5c1ae0"
          id="1st vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(16px)",
              }
            }
          />
        </button>
      </div>
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="2nd vertex"
        >
          2nd vertex
        </label>
        <button
          aria-checked={false}
          className="css-5c1ae0"
          id="2nd vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(0)",
              }
            }
          />
        </button>
      </div>
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="3th vertex"
        >
          3th vertex
        </label>
        <button
          aria-checked={false}
          className="css-5c1ae0"
          id="3th vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(0)",
              }
            }
          />
        </button>
      </div>
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="4th vertex"
        >
          4th vertex
        </label>
        <button
          aria-checked={true}
          className="css-5c1ae0"
          id="4th vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(16px)",
              }
            }
          />
        </button>
      </div>
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="5th vertex"
        >
          5th vertex
        </label>
        <button
          aria-checked={true}
          className="css-5c1ae0"
          id="5th vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(16px)",
              }
            }
          />
        </button>
      </div>
    </div>
    <div
      className="css-1jq4bbb"
    >
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="6th vertex"
        >
          6th vertex
        </label>
        <button
          aria-checked={false}
          className="css-5c1ae0"
          id="6th vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(0)",
              }
            }
          />
        </button>
      </div>
      <div
        className="css-vd9ji2"
      >
        <label
          className="css-1acae00"
          htmlFor="7th vertex"
        >
          7th vertex
        </label>
        <button
          aria-checked={false}
          className="css-5c1ae0"
          id="7th vertex"
          onClick={[Function]}
          role="switch"
          type="button"
        >
          <div
            aria-hidden={true}
            className="css-pxmhqy"
            style={
              Object {
                "transform": "translateX(0)",
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`chaos-game model renders properly 2`] = `
<canvas
  height={500}
  width={500}
/>
`;
